contracts/libraries/osmosis-cl-withdrawer/src/testing/test_suite.rs

```diff
@@ -0,0 +1,145 @@
+use cosmwasm_std::{coin, Coin, Int64};
+
+use osmosis_std::{
+    cosmwasm_to_proto_coins,
+    types::osmosis::concentratedliquidity::v1beta1::{MsgCreatePosition, MsgTransferPositions},
+};
+use osmosis_test_tube::{
+    osmosis_std::types::{
+        cosmwasm::wasm::v1::MsgExecuteContractResponse,
+        osmosis::concentratedliquidity::v1beta1::{UserPositionsRequest, UserPositionsResponse},
+    },
+    Account, ConcentratedLiquidity, ExecuteResponse, Module, Wasm,
+};
+use valence_library_utils::msg::{ExecuteMsg, InstantiateMsg};
+use valence_osmosis_utils::{
+    suite::{OsmosisTestAppBuilder, OsmosisTestAppSetup, OSMO_DENOM, TEST_DENOM},
+    testing::concentrated_liquidity::ConcentratedLiquidityPool,
+};
+
+use crate::msg::{FunctionMsgs, LibraryConfig, LibraryConfigUpdate};
+
+use super::ConcentratedLiquidityExts;
+
+// use super::ConcentratedLiquidityExt;
+
+pub struct LPerTestSuite {
+    pub inner: OsmosisTestAppSetup<ConcentratedLiquidityPool>,
+    pub lw_addr: String,
+    pub input_acc: String,
+    pub output_acc: String,
+}
+
+impl Default for LPerTestSuite {
+    fn default() -> Self {
+        Self::new(vec![
+            coin(1_000_000u128, OSMO_DENOM),
+            coin(1_000_000u128, TEST_DENOM),
+        ])
+    }
+}
+
+impl LPerTestSuite {
+    pub fn new(with_input_bals: Vec<Coin>) -> Self {
+        let inner: OsmosisTestAppSetup<ConcentratedLiquidityPool> =
+            OsmosisTestAppBuilder::new().build().unwrap();
+
+        // Create two base accounts
+        let wasm = Wasm::new(&inner.app);
+
+        let account_code_id = inner.store_account_contract();
+        let input_acc = inner.instantiate_input_account(account_code_id);
+        let output_acc = inner.instantiate_input_account(account_code_id);
```

Comment:
nit: bit confusing here, would prob rename function to instantiate_base_account

Reply:
that's a good one, also changed it for astroport